plugins/search-backend-module-explore/src/collators/ToolDocumentCollatorFactory.ts

```diff
@@ -87,7 +93,17 @@ export class ToolDocumentCollatorFactory implements DocumentCollatorFactory {
 
   private async fetchTools() {
     const baseUrl = await this.discovery.getBaseUrl('explore');
-    const response = await fetch(`${baseUrl}/tools`);
+
+    let headers = {};
+
+    if (this.tokenManager) {
+      const { token } = await this.tokenManager.getToken();
```

Comment:
It would be greatly appreciated if you could test this new behavior if you have any free time 🙂

Reply:
Hey @camilaibs we are running this code in our backstage installation and tested it there and it's working great for us. Should I test anything else or maybe a unit test? (might take me a bit, still learning Typescript!)

Reply:
Hey, I thought about a unit test, but only if you have a little extra time to try it out. If not, you can create a separate pull request further helping us to increase the code coverage. You're more than welcome to contribute again 🙂

Reply:
I don't have a ton of time at the moment but, I noted this down on my todo list to come back and add a test!